Add NGINX proxy for logs (#3893)
DataDog/documentation: content/agent/proxy.md

{{% /tab %}}
{{< /tabs >}}

## Using NGINX as a Proxy

[NGINX][3] is a web server which can also be used as a reverse proxy, load balancer, mail proxy, and HTTP cache. You can use NGINX as a proxy to send logs to Datadog:

`agent ---> nginx ---> Datadog` or  

> `external log shipper ---> nginx ---> Datadog`

Reviewer: Let's remove this one and focus on the use case were the agent is used (especially as the endpoint is different when not using the agent).

### Proxy forwarding with NGINX
#### NGINX configuration

This example `nginx.conf` can be used to proxy logs to the Datadog intake. It does TLS wrapping to ensure internal plaintext logs are encrypted between your proxy and Datadog's log intake API endpoint:

```
user nginx;
worker_processes auto;
error_log /var/log/nginx/error.log;
pid /run/nginx.pid; 

include /usr/share/nginx/modules/*.conf;
events {
    worker_connections 1024;
}
# TCP Proxy for Datadog logs
stream {
    server {
        listen 10514; #proxy listen port
        proxy_ssl on;
        proxy_pass agent-intake.logs.datadoghq.com:10516;
    }
}
```

> #### Shipper configuration

Reviewer: Let's change this by Agent configuration

{{< tabs >}}
{{% tab "Agent v6" %}}

To use the Datadog Agent as the logs collector, instruct the Agent to use the newly created proxy instead of establishing a connection directly with the logs intake by updating `datadog.yaml`:

```yaml
logs_config:

>   dd_url: myProxyServer.myDomain

Reviewer: now the proper configuration is `logs_dd_url: myProxyServer.myDomain:10514` instead of using `dd_url` and `dd_port`

  dd_port: 10514

>   dev_mode_use_proto: false

Reviewer: Let's remove this and keep the proto mode activated

  dev_mode_no_ssl: true
```

> The `dev_mode_no_ssl` parameter is set to `true` because establishing the SSL/TLS connection is handled by NGINX's `proxy_ssl on` option. **Note**: Set this option to `false` if you don't intend to use a proxy which can encrypt the connection to the logs intake.

Reviewer 2: Possibly
```suggestion
The `dev_mode_no_ssl` parameter is set to `true` as the SSL/TLS connection is established via NGINX's `proxy_ssl on` option. **Note**: Set this option to `false` if you don't intend to use a proxy which can encrypt the connection to the logs intake.
```

Author: Checking with @l0k0ms to see if `via` should be avoided for translation purposes.

{{% /tab %}}

> {{% tab "NXLogs" %}}

Reviewer: Let's remove any instruction which are not with the Agent

In Windows, NXLogs is frequently used as a logs collector. In the configuration, change the host and port in the `<Output>` block to those of your proxy instead of the Datadog intake.

**Example**:

```
<Output out>
  Module om_tcp
  Host myProxyServer.myDomain
  Port 10514
  Exec to_syslog_ietf();
  Exec $raw_event="<YOUR_DATADOG_API_KEY> "+$raw_event;
</Output>
```

{{% /tab %}}

> {{% tab "Fluentd" %}}

Reviewer: Let's remove instruction as not with the Agent
